src/lualdap.c

```diff
@@ -981,8 +981,12 @@ static int lualdap_open_simple (lua_State *L) {
 	if (strstr(host, "://") != NULL) {
 		err = ldap_initialize(&conn->ld, host);
 	} else {
-		const char *host_with_schema = lua_pushfstring(L, "ldap://%s", host);
-		err = ldap_initialize(&conn->ld, host_with_schema);
+		lua_getfield (L, 1, "gsub");
+		lua_pushvalue (L, 1); /* host */
+		lua_pushstring (L, "(%S+)");
+		lua_pushstring (L, "ldap://%1");
+		lua_call (L, 3, 1); /* host:gsub('(%S+)', 'ldap://%1') */
+		err = ldap_initialize(&conn->ld, lua_tostring (L, -1));
```

Comment:
I would add a comment that `ldap_initialize` handles comma and whitespace separated hostnames itself (i.e. we do not need to do anything on our side to make this work).

Reply:
Ok